core/services/ocr2/plugins/ocr2keeper/evmregistry/v21/logprovider/buffer_v1.go

```diff
@@ -105,7 +106,11 @@ func (b *logBuffer) Enqueue(uid *big.Int, logs ...logpoller.Log) (int, int) {
 		b.setUpkeepQueue(uid, buf)
 	}
 
-	latestLogBlock, uniqueBlocks := blockStatistics(logs...)
+	latestLogBlock, uniqueBlocks, reorgBlocks := b.blockStatistics(logs...)
```

Comment:
now that this function accesses and changes buffer state, it seems we need to acquire a lock

Reply:
discussed offline, we'll need to combine lock for blockStatistics and evictReorgdLogs (maybe combine those)
also would be good to think through concurrency of the whole function